src/main/javac/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java

```diff
@@ -198,7 +210,17 @@ private static boolean isTosca(Map<?,?> obj) {
                 String templateId = templateE.getKey();
                 NodeTemplate template = templateE.getValue();
 
-                EntitySpec<VanillaSoftwareProcess> thisNode = new ToscaComputeToVanillaConverter(mgmt).toSpec(templateId, template);
+                EntitySpec<? extends Entity> thisNode = null;
+                try {
+                    thisNode = EntitySpec.create((Class<Entity>) Class.forName(template.getType()));
```

Comment:
All nodes that we are parsing in the loop are everything except `tosca.nodes.Compute`. As A4C can send a list of "Brooklyn nodes" (i.e. Brooklyn entities imported through the plugin) the line 215 will attempt to resolve a Brooklyn entity based on its type.
Granted that it's not really secure and we should resolve that through the catalog.

I'll add a TODO comment for now and work on a cleaner solution.